mrp_bom_nested_bom_generator/models/mrp_nested_bom.py

```diff
@@ -0,0 +1,188 @@
+        related="bom_id.product_tmpl_id",
+    )
+
+    all_attribute_ids = fields.Many2many(
+        "product.attribute",
+        "all_attribute_ids_ref",
+        string="Parent Product Template attribute",
+        compute="_compute_all_attribute_ids",
+        store=True,
+    )
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product",
+        domain="[('id', '!=', parent_id)]",
+    )
+    product_qty = fields.Float(
+        string="Qty",
+        default=1,
+        required=True,
+    )
+    uom_id = fields.Many2one(
+        "uom.uom",
+        string="Unit of Measure",
+        related="product_tmpl_id.uom_id",
+        readonly=True,
+    )
+
+    attribute_ids = fields.Many2many(
+        "product.attribute",
+        "attribute_ids_ref",
+        string="Attributes",
+    )
+
+    @api.onchange("product_tmpl_id")
+    def _onchange_product_tmpl_id(self) -> None:
+        """
+        Change attribute field at changing product template field
+        :return None
+        """
+        self.attribute_ids = self.product_tmpl_id.attribute_line_ids.mapped(
+            "attribute_id"
+        )
+
+    @api.depends("bom_id.nested_bom_ids", "parent_id.attribute_line_ids")
+    def _compute_all_attribute_ids(self) -> None:
+        """
+        Compute product attributes by parent product and component
+        :return None
+        """
+        for rec in self:
+            default_attributes = rec.product_tmpl_id.attribute_line_ids.mapped(
+                "attribute_id"
+            )
+            parent_attributes = rec.parent_id.attribute_line_ids.mapped("attribute_id")
+            attributes_ids = default_attributes | parent_attributes
+            rec.all_attribute_ids = attributes_ids
+
+    def write(self, vals):
+        self.mapped("bom_id").write({"changed_nested_bom": True})
+        return super(MrpNestedBomLine, self).write(vals)
+
+    @api.model
+    def create_product(self, bom_id: int) -> int:
+        """
+        Create product by parent product
+        :param int bom_id: mrp.bom record id
+        :return new product template id
+        :rtype int
+        """
+        product_template = self.env["mrp.bom"].browse(bom_id).product_tmpl_id
+        nested_bom_count: int = self.search_count(
+            [("parent_id", "=", product_template.id)]
+        )
+        return (
+            self.env["product.template"]
+            .create(
+                {"name": "{} #{}".format(product_template.name, nested_bom_count + 1)}
+            )
+            .id
+        )
+
+    @api.model
+    def create(self, vals: Dict[str, Any]):
+        product_tmpl_id = vals.get("product_tmpl_id", False)
+        product_qty = vals.get("product_qty", False)
+        bom_id = vals.get("bom_id", False)
+
+        # Create product template if product_tmpl_id not set
+        if not product_tmpl_id:
+            vals.update(product_tmpl_id=self.create_product(bom_id))
```

Comment:
This will crash in case of bom_id is False. Is this case covered by tests?

Reply:
Resolved